lambdas/services/create_document_reference_service.py

```diff
@@ -103,6 +102,15 @@ def create_document_reference_request(
                 )
 
             if arf_documents:
+                incomplete_arf_upload_records = (
+                    self.fetch_incomplete_arf_upload_records(nhs_number)
+                )
+
+                self.stop_if_upload_is_in_process(incomplete_arf_upload_records)
+                self.remove_records_of_failed_upload(
+                    self.arf_dynamo_table, incomplete_arf_upload_records
+                )
+
                 self.create_reference_in_dynamodb(
                     self.arf_dynamo_table, arf_documents_dict_format
                 )
```

Comment:
If it's ok, could you move this to a separated function? this `create_document_reference_request` is getting too big.

Reply:
Thanks, got this in a separate function now :)